src/Aspire.Hosting/Pipelines/DistributedApplicationPipeline.cs

```diff
@@ -137,12 +137,43 @@ public async Task ExecuteAsync(PipelineContext context)
 
         ValidateSteps(allSteps);
 
+        // Convert RequiredBy relationships to DependsOn relationships before filtering
+        var allStepsByName = allSteps.ToDictionary(s => s.Name, StringComparer.Ordinal);
+        NormalizeRequiredByToDependsOn(allSteps, allStepsByName);
+
         var (stepsToExecute, stepsByName) = FilterStepsForExecution(allSteps, context);
 
         // Build dependency graph and execute with readiness-based scheduler
         await ExecuteStepsAsTaskDag(stepsToExecute, stepsByName, context).ConfigureAwait(false);
     }
 
+    /// <summary>
+    /// Converts all RequiredBy relationships to their equivalent DependsOn relationships.
+    /// If step A is required by step B, this adds step A as a dependency of step B.
+    /// </summary>
+    private static void NormalizeRequiredByToDependsOn(
+        List<PipelineStep> steps,
+        Dictionary<string, PipelineStep> stepsByName)
+    {
+        foreach (var step in steps)
+        {
+            foreach (var requiredByStep in step.RequiredBySteps)
+            {
+                if (!stepsByName.TryGetValue(requiredByStep, out var requiredByStepObj))
```

Comment:
This should be an error, right? Who is verifying that `requiredByStep` is an actual step?